File: backend/src/main/java/com/staffly/backend/employee/Employee.java
package com.staffly.backend.employee;

import com.staffly.backend.branch.Branch;
import com.staffly.backend.tenant.TenantAwareEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import org.hibernate.annotations.Filter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Entity
@Table(name = "employee")
@Filter(name = "tenantFilter", condition = "company_id = :companyId")
public class Employee extends TenantAwareEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;

    @Column(name = "nombre", nullable = false)
    private String nombre;

    @Column(name = "apellido", nullable = false)
    private String apellido;

    @Column(name = "documento", nullable = false)
    private String documento;

    @Column(name = "fecha_nacimiento", nullable = false)
    private LocalDate fechaNacimiento;

    @Column(name = "fecha_ingreso", nullable = false)
    private LocalDate fechaIngreso;

    @Column(name = "fecha_egreso")
    private LocalDate fechaEgreso;

    @Enumerated(EnumType.STRING)
    @Column(name = "tipo_contrato", nullable = false)
    private TipoContrato tipoContrato;

    @Column(name = "categoria", nullable = false)
    private String categoria;

    @Column(name = "sueldo_base", nullable = false)
    private BigDecimal sueldoBase;

    @Column(name = "telefono")
    private String telefono;

    @Column(name = "email_contacto")
    private String emailContacto;

    @Enumerated(EnumType.STRING)
    @Column(name = "estado_laboral", nullable = false)
    private EstadoLaboral estadoLaboral = EstadoLaboral.ACTIVO;

    @Enumerated(EnumType.STRING)
    @Column(name = "estado_liquidacion", nullable = false)
    private EstadoLiquidacion estadoLiquidacion = EstadoLiquidacion.AL_DIA;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "employee_branch",
            joinColumns = @JoinColumn(name = "employee_id"),
            inverseJoinColumns = @JoinColumn(name = "branch_id")
    )
    private Set<Branch> branches = new HashSet<>();

    public UUID getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public String getDocumento() {
        return documento;
    }

    public void setDocumento(String documento) {
        this.documento = documento;
    }

    public LocalDate getFechaNacimiento() {
        return fechaNacimiento;
    }

    public void setFechaNacimiento(LocalDate fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }

    public LocalDate getFechaIngreso() {
        return fechaIngreso;
    }

    public void setFechaIngreso(LocalDate fechaIngreso) {
        this.fechaIngreso = fechaIngreso;
    }

    public LocalDate getFechaEgreso() {
        return fechaEgreso;
    }

    public void setFechaEgreso(LocalDate fechaEgreso) {
        this.fechaEgreso = fechaEgreso;
    }

    public TipoContrato getTipoContrato() {
        return tipoContrato;
    }

    public void setTipoContrato(TipoContrato tipoContrato) {
        this.tipoContrato = tipoContrato;
    }

    public String getCategoria() {
        return categoria;
    }

    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }

    public BigDecimal getSueldoBase() {
        return sueldoBase;
    }

    public void setSueldoBase(BigDecimal sueldoBase) {
        this.sueldoBase = sueldoBase;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getEmailContacto() {
        return emailContacto;
    }

    public void setEmailContacto(String emailContacto) {
        this.emailContacto = emailContacto;
    }

    public EstadoLaboral getEstadoLaboral() {
        return estadoLaboral;
    }

    public void setEstadoLaboral(EstadoLaboral estadoLaboral) {
        this.estadoLaboral = estadoLaboral;
    }

    public EstadoLiquidacion getEstadoLiquidacion() {
        return estadoLiquidacion;
    }

    public void setEstadoLiquidacion(EstadoLiquidacion estadoLiquidacion) {
        this.estadoLiquidacion = estadoLiquidacion;
    }

    public Set<Branch> getBranches() {
        return branches;
    }
}

File: frontend/src/app/features/employees/models/employee.ts
export type EstadoLaboral = 'ACTIVO' | 'LICENCIA' | 'SUSPENDIDO' | 'BAJA';
export type EstadoLiquidacion = 'AL_DIA' | 'PENDIENTE';
export type TipoContrato = 'JORNADA_COMPLETA' | 'JORNADA_PARCIAL' | 'POR_HORA';

export interface Employee {
  id: string;
  nombre: string;
  apellido: string;
  documento: string;
  fechaNacimiento: string;
  fechaIngreso: string;
  fechaEgreso: string | null;
  tipoContrato: TipoContrato;
  categoria: string;
  sueldoBase: number;
  telefono: string | null;
  emailContacto: string | null;
  estadoLaboral: EstadoLaboral;
  estadoLiquidacion: EstadoLiquidacion;
  branchIds: string[];
}

export interface CreateEmployeeRequest {
  nombre: string;
  apellido: string;
  documento: string;
  fechaNacimiento: string;
  fechaIngreso: string;
  fechaEgreso?: string;
  tipoContrato: TipoContrato;
  categoria: string;
  sueldoBase: number;
  telefono?: string;
  emailContacto?: string;
  branchIds: string[];
}

export interface UpdateEmployeeRequest {
  nombre?: string;
  apellido?: string;
  documento?: string;
  fechaNacimiento?: string;
  fechaIngreso?: string;
  fechaEgreso?: string;
  tipoContrato?: TipoContrato;
  categoria?: string;
  sueldoBase?: number;
  telefono?: string;
  emailContacto?: string;
  branchIds?: string[];
}

export interface UpdateEmployeeStatusRequest {
  estadoLaboral: EstadoLaboral;
}

export interface EmployeeListFilters {
  estadoLaboral?: EstadoLaboral;
  branchId?: string;
  search?: string;
}
